File: microservice-IANucle-ForeIA/src/utils/authJWTUtil.py
import jwt
import os
from dotenv import load_dotenv

load_dotenv()

class AuthJWTUtil:
    
    def __init__(self):
        self.jwtKey = os.getenv("FOREIANUCLE_MICROSERVICE_TOKENKEY")
        
    def verifyToken(self, token):
        try:
            payload = jwt.decode(token, self.jwtKey,algorithms=["HS256"])
                
            return payload

        except Exception as error:
            raise ValueError("ivalid token")
        
    
    
    
    
authJWTUtil = AuthJWTUtil()

File: back-ForeIA/src/utils/AuthJWTUtil.js
import jwt from "jsonwebtoken";



class AuthJWTUtil {

    constructor() {
        this.tokenOptions = {
            algorithm: 'HS256',
            expiresIn: '72h',
            issuer: "api.tomasperez.work"
        };
    }

    singToken(payload) {

        const user = {

            _id: payload._id.toString(),
            username: payload.username,
            email: payload.email

        }

        const token = jwt.sign(user, process.env.JWT_KEY, this.tokenOptions)

        return token

    }

    verifyToken(token) {

        try {

            const tokenDecoded = jwt.verify(token, process.env.JWT_KEY, this.tokenOptions)

            return tokenDecoded;

        } catch (error) {

            throw error;

        }

    }

    foreIANucleMicroserviceTokens() {

        const tokenOptionsForForeIANucle = {

            algorithm: 'HS256',
            expiresIn: "72h",
            issuer: "api.tomasperez.work"

        }

        return jwt.sign({}, process.env.FOREIANUCLEMICROSERVICE_TOKENKEY, tokenOptionsForForeIANucle)

    }

}

export const authJWTUtil = new AuthJWTUtil();